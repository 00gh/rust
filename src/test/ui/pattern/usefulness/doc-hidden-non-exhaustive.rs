// aux-build:hidden.rs

extern crate hidden;

use hidden::HiddenEnum;

fn main() {
    match HiddenEnum::A {
        HiddenEnum::A => {}
        HiddenEnum::B => {}
    }
    //~^^^^ non-exhaustive patterns: `_` not covered

    match HiddenEnum::A {
        HiddenEnum::A => {}
        HiddenEnum::C => {}
    }
    //~^^^^ non-exhaustive patterns: `B` not covered

    match HiddenEnum::A {
        HiddenEnum::A => {}
    }
    //~^^^ non-exhaustive patterns: `B` and `_` not covered

    match None {
        None => {}
        Some(HiddenEnum::A) => {}
    }
    //~^^^^ non-exhaustive patterns: `Some(B)` and `Some(_)` not covered
}
