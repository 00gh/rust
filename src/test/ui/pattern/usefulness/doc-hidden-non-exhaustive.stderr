error[E0004]: non-exhaustive patterns: `_` not covered
  --> $DIR/doc-hidden-non-exhaustive.rs:8:11
   |
LL |     match HiddenEnum::A {
   |           ^^^^^^^^^^^^^ pattern `_` not covered
   |
   = help: ensure that all possible cases are being handled, possibly by adding wildcards or more match arms
   = note: the matched value is of type `HiddenEnum`

error[E0004]: non-exhaustive patterns: `B` not covered
  --> $DIR/doc-hidden-non-exhaustive.rs:14:11
   |
LL |     match HiddenEnum::A {
   |           ^^^^^^^^^^^^^ pattern `B` not covered
   |
note: `Foo` defined here
  --> $DIR/auxiliary/hidden.rs:3:5
   |
LL |     B,
   |     - not covered
   |
   = help: ensure that all possible cases are being handled, possibly by adding wildcards or more match arms
   = note: the matched value is of type `HiddenEnum`

error[E0004]: non-exhaustive patterns: `B` and `_` not covered
  --> $DIR/doc-hidden-non-exhaustive.rs:20:11
   |
LL |     match HiddenEnum::A {
   |           ^^^^^^^^^^^^^ patterns `B` and `_` not covered
   |
note: `Foo` defined here
  --> $DIR/auxiliary/hidden.rs:3:5
   |
LL |     B,
   |     - not covered
   |
   = help: ensure that all possible cases are being handled, possibly by adding wildcards or more match arms
   = note: the matched value is of type `HiddenEnum`

error[E0004]: non-exhaustive patterns: `Some(B)` and `Some(_)` not covered
  --> $DIR/doc-hidden-non-exhaustive.rs:25:11
   |
LL |     match None {
   |           ^^^^ patterns `Some(B)` and `Some(_)` not covered
   |
note: `Option<Foo>` defined here
  --> $SRC_DIR/core/src/option.rs:LL:COL
   |
LL |     Some(#[stable(feature = "rust1", since = "1.0.0")] T),
   |     ---- not covered
   |
   = help: ensure that all possible cases are being handled, possibly by adding wildcards or more match arms
   = note: the matched value is of type `Option<HiddenEnum>`

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0004`.
